packages/quantic/force-app/main/default/lwc/quanticResultText/quanticResultText.js

```diff
@@ -0,0 +1,78 @@
+import {LightningElement, api} from 'lwc';
+
+/** @typedef {import("coveo").Result} Result */
+
+/**
+ * The `QuanticResultText` component displays a given result field value.
+ * @category Result Template
+ * @example
+ * <c-quantic-result-text result={result} label="Source" field="source"></c-quantic-result-text>
+ */
+export default class QuanticResultText extends LightningElement {
+  /**
+   * The [result item](https://docs.coveo.com/en/headless/latest/reference/search/controllers/result-list/#result) to use.
+   * @api
+   * @type {Result}
+   */
+  @api result;
+  /**
+   * (Optional) The label to display.
+   * @api
+   * @type {string}
+   * @defaultValue `undefined`
+   */
+  @api label;
+  /**
+   * The field whose values you want to display.
+   * @api
+   * @type {string}
+   */
+  @api field;
+  /**
+   * The function used to format the displayed value.
+   * @api
+   * @type {Function}
+   * @param {string} value
+   * @returns {string}
+   */
+  @api formattingFunction;
+
+  error;
+
+  connectedCallback() {
+    if (!this.result || !this.field) {
+      console.error(
+        `The ${this.template.host.localName} requires a result and a field to be specified.`
+      );
+      this.error = `${this.template.host.localName} Error`;
+    }
+  }
+
+  /**
+   * The value of the given result field.
+   * @returns {string | undefined}
+   */
+  get fieldValue() {
+    // @ts-ignore
+    return this.result.raw[this.field];
+  }
+
+  /**
+   * Whether the given result has a value for the given field
+   * @returns {boolean}
+   */
+  get hasFieldValue() {
+    return !!this.fieldValue.length;
```

Comment:
This statement will fail if the value is not a string. If, for some reason, the field value is not of the expected type, the component should fail gracefully. It could raise a warning and make sure `hasFieldValue` returns `false` in such case.

Reply:
You're right. I'm in the process of adding the bueno library as a quantic dependency to do better input validation. that commit is coming soon.

Reply:
and some optional chaining couldn't hurt